regulation/tree.py

```diff
@@ -322,9 +322,15 @@ def build_internal_citations_layer(root):
         for cite, positions in cite_positions.items():
             # positions = find_all_occurrences(par_text, text)
             for pos in positions:
-                #print cite, positions, par_label
-                cite_dict = {'citation': cite_targets[cite],
-                             'offsets': [[pos, pos + len(cite)]]}
+                # Handle empty citations
+                try:
+                    cite_dict = {'citation': cite_targets[cite],
+                                 'offsets': [[pos, pos + len(cite)]]}
+                except TypeError as e:
+                    print "TypeError occurred: {}".format(str(e))
+                    print "Look for an empty citation in {} @ pos {}".format(par_label, positions)
```

Comment:
I really do like this error though, I'd like use to use that pattern in other places!